### PR TITLE
refactoring extract-stack

### LINKED ISSUE #142: [TOOL-56] Rewrite queueHandler so we can use a single queue for the whole Extract stack
<sub>From [SyncLinear.com](https://synclinear.com) | [TOOL-56](https://linear.app/crocoder/issue/TOOL-56/rewrite-queuehandler-so-we-can-use-a-single-queue-for-the-whole)</sub>

### PR DESCRIPTION
**Pull request by davidabram**
closes #142

### CODE REVIEW

File: apps/extract-stack/src/extract-merge-request-commits.ts [hunk 1]
```diff
@@ -4,23 +4,52 @@ import { Clerk } from "@clerk/clerk-sdk-node";
...
-import { extractMergeRequestCommitMessage, type extractMergeRequestData } from "./messages";
+import { createMessageHandler } from "./create-message";
+import { MessageKind, metadataSchema } from "./messages";
+import { z } from "zod";
 
-const clerkClient = Clerk({ secretKey: Config.CLERK_SECRET_KEY });
-const client = createClient({
-  url: Config.DATABASE_URL,
-  authToken: Config.DATABASE_AUTH_TOKEN,
+export const mrcsh = createMessageHandler({
```
Comment (davidabram) [inline apps/extract-stack/src/extract-merge-request-commits.ts:14]: MergeRequestCommitMessageHandler is too long and the stack fails to build for some reason.

File: apps/extract-stack/src/extract-merge-request-diffs.ts [hunk 2]
```diff
@@ -5,12 +5,36 @@ import { GitHubSourceControl, GitlabSourceControl } from "@acme/source-control";
...
 import { extractMergeRequestsEvent } from "./events";
-import { extractMergeRequestDiffMessage } from "./messages";
-import type { extractMergeRequestData } from "./messages";
-import { QueueHandler } from "./create-message";
+import { createMessageHandler } from "./create-message";
+import { metadataSchema } from "./messages";
+import { z } from "zod";
+
+export const mergeRequestDiffSenderHandler = createMessageHandler({
+  kind: 'merge-request-diff',
```
Comment (Ante-Koceic) [inline apps/extract-stack/src/extract-merge-request-diffs.ts:16]: Use MessageKind.MergeRequestDiff instead of 'merge-request-diff'

**Review by ivke995 — APPROVED** [review]

**Review by Ante-Koceic — APPROVED** [review]

**Pull request merged**